source/slang/slang-ast-print.cpp

```diff
@@ -74,6 +74,1052 @@ void ASTPrinter::addType(Type* type)
+            break;
+        case BaseType::Int8:
+            sb << "b";
+            break;
+        case BaseType::UInt8:
+            sb << "ub";
+            break;
+        default:
+            // Don't add a suffix for other types
+            break;
+        }
+    }
+    else if (const auto floatLit = as<FloatingPointLiteralExpr>(expr))
+    {
+        sb << floatLit->value;
+        // Handle suffix types without using getBaseTypeName
+        switch (floatLit->suffixType)
+        {
+        case BaseType::Float:
+            sb << "f";
+            break;
+        case BaseType::Double:
+            // No suffix for default double
+            break;
+        case BaseType::Half:
+            sb << "h";
+            break;
+        default:
+            // Don't add a suffix for other types
+            break;
+        }
+    }
+    else if (const auto boolLit = as<BoolLiteralExpr>(expr))
+    {
+        sb << (boolLit->value ? "true" : "false");
+    }
+    else if (as<NullPtrLiteralExpr>(expr))
+    {
+        sb << "nullptr";
+    }
+    else if (as<NoneLiteralExpr>(expr))
+    {
+        sb << "none";
+    }
+    else if (const auto stringLit = as<StringLiteralExpr>(expr))
+    {
+        sb << "\"" << stringLit->value << "\"";
+    }
+    else if (const auto initList = as<InitializerListExpr>(expr))
+    {
+        sb << "{";
+        bool first = true;
+        for (auto arg : initList->args)
+        {
+            if (!first)
+                sb << ", ";
+            addExpr(arg);
+            first = false;
+        }
+        sb << "}";
+    }
+    else if (const auto arrayLengthExpr = as<GetArrayLengthExpr>(expr))
+    {
+        if (arrayLengthExpr->arrayExpr)
+        {
+            addExpr(arrayLengthExpr->arrayExpr);
+            sb << ".Length";
+        }
+        else
+        {
+            sb << "<unknown-array-length>";
+        }
+    }
+    else if (const auto expandExpr = as<ExpandExpr>(expr))
+    {
+        sb << "...";
+        if (expandExpr->baseExpr)
+        {
+            addExpr(expandExpr->baseExpr);
+        }
+    }
+    else if (const auto eachExpr = as<EachExpr>(expr))
+    {
+        sb << "each ";
+        if (eachExpr->baseExpr)
+        {
+            addExpr(eachExpr->baseExpr);
+        }
+    }
+    else if (const auto aggTypeCtorExpr = as<AggTypeCtorExpr>(expr))
+    {
+        addType(aggTypeCtorExpr->base.type);
+        sb << "(";
+        bool first = true;
+        for (auto arg : aggTypeCtorExpr->arguments)
+        {
+            if (!first)
+                sb << ", ";
+            addExpr(arg);
+            first = false;
+        }
+        sb << ")";
+    }
+    else if (const auto invokeExpr = as<InvokeExpr>(expr))
+    {
+        if (const auto operatorExpr = as<OperatorExpr>(invokeExpr))
+        {
+            if (const auto infixExpr = as<InfixExpr>(operatorExpr))
+            {
+                // Binary operator
+                if (invokeExpr->arguments.getCount() == 2)
```

Comment:
This doesn't seem to handle insertion of () to properly express operator priorities. We could just emit () for all cases for now.